common/static/common/js/discussion/utils.js

```diff
@@ -151,61 +153,59 @@
         };
 
         DiscussionUtil.discussionAlert = function(header, body) {
-            var alertDiv, alertTrigger;
+            var $alertDiv, $alertTrigger;
             if ($("#discussion-alert").length === 0) {
-                alertDiv = $(
-                    "<div class='modal' role='alertdialog' id='discussion-alert' " +
-                    "aria-describedby='discussion-alert-message'/>"
+                $alertDiv = $(
+                    edx.HtmlUtils.joinHtml(
+                        edx.HtmlUtils.HTML("<div class='modal' role='alertdialog' id='discussion-alert' "),
+                        edx.HtmlUtils.HTML("aria-describedby='discussion-alert-message'/>")
+                    ).toString()
                 ).css("display", "none");
-                alertDiv.html(
-                    "<div class='inner-wrapper discussion-alert-wrapper'>" +
-                    "   <button class='close-modal dismiss' title='" + gettext("Close") + "'>" +
-                    "       <span class='icon fa fa-times' aria-hidden='true'></span>" +
-                    "   </button>" +
-                    "   <header><h2/><hr/></header>" +
-                    "   <p id='discussion-alert-message'/><hr/>" +
-                    "   <button class='dismiss'>" + gettext("OK") + "</button>" +
-                    "</div>"
+                edx.HtmlUtils.append(
+                    $alertDiv,
+                    edx.HtmlUtils.joinHtml(
+                        edx.HtmlUtils.HTML("<div class='inner-wrapper discussion-alert-wrapper'>"),
+                            edx.HtmlUtils.interpolateHtml(
+                                edx.HtmlUtils.HTML("<button class='close-modal dismiss' title='{title}'>"),
+                                {title: gettext("Close")}
+                            ),
+                                edx.HtmlUtils.HTML("<span class='icon fa fa-times' aria-hidden='true'></span>"),
+                            edx.HtmlUtils.HTML("</button>"),
+                            edx.HtmlUtils.HTML("<header><h2/><hr/></header>"),
+                            edx.HtmlUtils.HTML("<p id='discussion-alert-message'/><hr/>"),
+                            edx.HtmlUtils.HTML("<button class='dismiss'>"),
+                                gettext("OK"),
+                            edx.HtmlUtils.HTML("</button>"),
+                        edx.HtmlUtils.HTML("</div>")
```

Comment:
I'm against HTML living inside Javascript generally; HTML should go in an Underscore template and be loaded into Javascript with RequireJSText, if that's an option (it may not be here, it looks like this code isn't RequireJS-ified yet).

That said, of the two patterns for writing HTML inside JS, I prefer the one @robrap linked to instead of the one that's shown here - if there is HTML in the JS, it should at least be readable as a multiline block of plain HTML, and not interspersed with JS on each line. I don't see the pattern above as much of a win for readability - it repeats itself frequently and removes the indentation of the HTML, which is key to understanding its structure and purpose. Perhaps this change was done for a different reason though?